### PR TITLE
Add: BigLake API Catalog Resource 

### PR DESCRIPTION
**Pull request by nevzheng**
Add `google_biglake_catalog`

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ x [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_biglake_catalog`
```

### CODE REVIEW

**Comment by nevzheng** [comment]
All PRs For BigLake APIs

github.com/GoogleCloudPlatform/magic-modules/pull/8751
github.com/GoogleCloudPlatform/magic-modules/pull/8753
github.com/GoogleCloudPlatform/magic-modules/pull/8754

**Comment by ScottSuarez** [comment]
Blocked on ga release

**Comment by nevzheng** [comment]
Thanks for asking about this series of PRs. #8751, #8753, #8754, supporting the (a) BigLake API the (a) BigLake API is different from 

(b) `BigLakeConfiguration" and "BigLakeManagedTables".

We are having a separate discussion on (b) regarding EAP, and readiness for inclusion in the ga and beta providers. I'm pretty sure (a) is GA. and we should be able to proceed with (a). Let me sync w/ you in chat to figure this out.

**Review by ScottSuarez — APPROVED** [review]

**Comment by ScottSuarez** [comment]
Thanks @nevzheng :) merged

File: mmv1/products/bigquerybiglake/product.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+!ruby/object:Api::Product
+name: Biglake
+display_name: Biglake
```
Comment (SarahFrench) [inline mmv1/products/bigquerybiglake/product.yaml:17]: I'm wondering if these lines should be prefixing the product name with BigQuery, similar to what's present in [`products/bigqueryanalyticshub/product.yaml`](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/bigqueryanalyticshub/product.yaml), `/products/bigqueryconnection/product.yaml` and other BigQuery-related products.

This change would mean that the package for this service would match the pattern for other BigQuery-related services, and also would mean that the provider's documentation would be listed in the sidebar nav as part of this block of similarly-named categories:

Screenshot 2023-08-29 at 12 55 53

Comment (SarahFrench) [reply mmv1/products/bigquerybiglake/product.yaml:17]: Discussed within the team and I'm going to change the folder name to `mmv1/products/biglake`

**Pull request merged**